main.js

```diff
@@ -0,0 +1,85 @@
+const inViewport = require('in-viewport');
+
+//find all convo cards
+const convoCards = document.getElementsByClassName('conversation');
+
+//for each card, define inViewport callback function
+[].slice.call(convoCards).forEach(card => inViewport(card, { offset: -200 }, function() {
+
+    //get card's classes so it can be used as a querying string
+    const cardSelector = card.className.split(" ").join(".");
+
+    //find all chat bubbles inside card
+    const chatList = document.querySelectorAll(`.${cardSelector} .bubble`);
+
+    //for each bubble, apply animation delay as indicated by element's data-delay attribute
+    [].slice.call(chatList).forEach(text => {
+      text.style.animationDelay = `${text.dataset.delay}ms`;
+      text.style.animationPlayState = 'running';
+
+      //adding 'finished' class will cause bubble's spinner to hide and bubble's text to appear
+      setTimeout(function(){
+        text.className += " finished";
+      }, `${1000 + Number(text.dataset.delay)}`)
+    })
```

Comment:
If the user resizes the page and triggers the breakpoint again within the delay, will this fire excessively?